sensor/kubernetes/listener/resources/selector_test.go

```diff
@@ -0,0 +1,215 @@
+		givenMatchingLabels                         map[string]string
+		expectedMatch                               bool
+		expectedInternalSelectorMatchesFunctionCall bool
+	}{
+		"Disjunction of empty list that should match nothing and labels with three elements": {
+			givenSelectorLabels: []map[string]string{labelsEmpty, labelsThreeElements},
+			matchEmptySelector:  []selectorWrapOption{emptyMatchesNothing(), emptyMatchesEverything()},
+			givenMatchingLabels: labelsOneElement,
+			expectedMatch:       false,
+			expectedInternalSelectorMatchesFunctionCall: false,
+		},
+		"Disjunction of empty list that should match everything and labels with three elements": {
+			givenSelectorLabels: []map[string]string{labelsEmpty, labelsThreeElements},
+			matchEmptySelector:  []selectorWrapOption{emptyMatchesEverything(), emptyMatchesEverything()},
+			givenMatchingLabels: labelsOneElement,
+			expectedMatch:       true,
+			expectedInternalSelectorMatchesFunctionCall: false,
+		},
+		"Disjunction of two selectors with more labels than the input": {
+			givenSelectorLabels: []map[string]string{labelsFiveElements, labelsThreeElements},
+			matchEmptySelector:  []selectorWrapOption{emptyMatchesNothing(), emptyMatchesNothing()},
+			givenMatchingLabels: labelsOneElement,
+			expectedMatch:       false,
+			expectedInternalSelectorMatchesFunctionCall: false,
+		},
+		"Disjunction of two selectors, one more labels than the input, one equal, without match": {
+			givenSelectorLabels: []map[string]string{labelsFiveElements, labelsThreeElements},
+			matchEmptySelector:  []selectorWrapOption{emptyMatchesNothing(), emptyMatchesNothing()},
+			givenMatchingLabels: labelsThreeElements2,
+			expectedMatch:       false,
+			expectedInternalSelectorMatchesFunctionCall: true,
+		},
+		"Disjunction of two selectors, one more labels than the input, one equal, with match": {
+			givenSelectorLabels: []map[string]string{labelsFiveElements, labelsThreeElements},
+			matchEmptySelector:  []selectorWrapOption{emptyMatchesNothing(), emptyMatchesNothing()},
+			givenMatchingLabels: labelsThreeElements,
+			expectedMatch:       true,
+			expectedInternalSelectorMatchesFunctionCall: true,
+		},
+		"Disjunction of two selectors, with less labels than the input": {
+			givenSelectorLabels: []map[string]string{labelsThreeElements2, labelsThreeElements},
+			matchEmptySelector:  []selectorWrapOption{emptyMatchesNothing(), emptyMatchesNothing()},
+			givenMatchingLabels: labelsFiveElements,
+			expectedMatch:       false,
+			expectedInternalSelectorMatchesFunctionCall: true,
+		},
+	}
+	for name, tt := range tests {
+		s.Run(name, func() {
+			s.hasMatchesBeenCalled = false
+			var selectorWrappers []selector
+			for i := 0; i < len(tt.givenSelectorLabels); i++ {
+				newSelector := createSelector(tt.givenSelectorLabels[i], tt.matchEmptySelector[i])
+				s.injectMockSelector(&newSelector)
+				selectorWrappers = append(selectorWrappers, newSelector)
+			}
```

Comment:
Maybe let's drop the C-style loops if we can

```suggestion
			for i, label :=  range tt.givenSelectorLabels {
				newSelector := createSelector(label, tt.matchEmptySelector[i])
				s.injectMockSelector(&newSelector)
				selectorWrappers = append(selectorWrappers, newSelector)
			}
```

Reply:
I usually use loops over ranges when I don't care about the index, but in this case the index would still be necessary in order to iterate over matchEmptySelector. My preference in cases like this one is C-style, but I'll change it to conform.